### PR TITLE
Added TP2-to-TP3 GraphML XSLT to resources

### PR DESCRIPTION
**Pull request by joshsh**
In gremlin-core as suggested by @spmallette

### CODE REVIEW

**Comment by okram** [comment]
Can you please make corresponding JIRA tickets and use the PR naming convention for your PRs? Also, this PR will need documentation as well as an update to the CHANGELOG. Finally, I don't know why this is targeted to `master/` when it seems it can go into `tp31/`.

**Comment by spmallette** [comment]
@joshsh - thanks - note that we still support 3.1.x (mostly for maintenance) on the tp31 branch, 3.2.x on the tp32 branch (for maintenance and non-breaking new features) and then do all dev for the future 3.3.x on master (which is all new stuff including breaking changes). In this case, I don't think it would hurt to put this on tp31 as @okram suggests - it might be a useful little bit of functionality for someone still on  3.1.x and shouldn't cause any conflicts. 

Is there an easy test you could write that just exercises the XSLT a little bit? Perhaps you could just put something in `gremlin-test` in `IoTest` as we have tests in there like `shouldReadLegacyGraphSON` which would be similar to what you have here. Maybe just copy that model:

https://github.com/apache/tinkerpop/blob/492c8e160743bb6d15bc07262a47275d18242eed/gremlin-test/src/main/java/org/apache/tinkerpop/gremlin/structure/io/IoTest.java#L438-L452

and use your XSLT instead of the `LegacyGraphSONReader`?

**Comment by joshsh** [comment]
Thanks for the info, @spmallette.  I will also look harder at graph migration as in IoGraphTest.  Note that I do not appear to have the create-issue permission in JIRA.

**Comment by spmallette** [comment]
Anyone who has an account can create issues. In fact, I just checked and I think you should even have right such that you can assign them to yourself (i see your account there as "joshsh"). Can you please check again?

**Comment by pluradj** [comment]
Created https://issues.apache.org/jira/browse/TINKERPOP-1608

File: gremlin-core/src/main/resources/tp2-to-tp3-graphml.xslt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    </xsl:template>
+
+    <xsl:template match="graphml:edge">
+        <edge id="{@id}" source="{@source}" target="{@target}">
+            <data key="labelE">
+                <xsl:value-of select="@label"/>
+            </data>
+        </edge>
+    </xsl:template>
+
```
Comment (pluradj) [inline gremlin-core/src/main/resources/tp2-to-tp3-graphml.xslt]: Appears that this transform creates the `labelE` child node, but the existing edge properties are not carried over.

File: gremlin-core/src/main/resources/tp2-to-tp3-graphml.xslt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            <xsl:apply-templates select="node()|@*"/>
+            <data key="labelV">vertex</data>
+        </node>
+    </xsl:template>
+
+    <xsl:template match="graphml:edge">
+        <edge id="{@id}" source="{@source}" target="{@target}">
+            <data key="labelE">
+                <xsl:value-of select="@label"/>
+            </data>
```
Comment (pluradj) [inline gremlin-core/src/main/resources/tp2-to-tp3-graphml.xslt]: Adding `<xsl:copy-of select="node()"/>` here seems to work.

Comment (joshsh) [inline gremlin-core/src/main/resources/tp2-to-tp3-graphml.xslt]: Thanks for catching that.  My test file had no edge properties.

**Comment by spmallette** [comment]
Closed in favor of #595

**Pull request closed without merging**